nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/main/java/org/nuxeo/ecm/platform/userworkspace/core/service/AbstractUserWorkspaceImpl.java

```diff
@@ -464,4 +446,9 @@ protected abstract DocumentModel doCreateUserWorkspace(CoreSession unrestrictedS
     protected abstract DocumentModel initCreateUserWorkspace(CoreSession unrestrictedSession, DocumentModel doc,
             String username);
 
+    @Override
+    public void invalidate() {
+        this.targetDomainName = null;
```

Comment:
`this.` is not needed.
Also make the `targetDomainName` field `volatile` because it's checked and modified and invalidated from different threads.